test/good/asyncapi-very-complex-schema-output.json

```diff
@@ -0,0 +1,876 @@
+              },
+              "maintenance": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "year": {
+                      "type": "number",
+                      "x-parser-schema-id": "<anonymous-schema-4>"
+                    },
+                    "cost": {
+                      "type": "number",
+                      "x-parser-schema-id": "<anonymous-schema-5>"
+                    }
+                  },
+                  "x-parser-schema-id": "streetLightMaintenance",
+                  "x-parser-schema-id-level": 3
+                },
+                "x-parser-schema-id": "streetLightMaintenanceArray",
+                "x-parser-schema-id-level": 2
+              },
+              "holder": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "country": {
+                      "type": "string",
+                      "x-parser-schema-id": "<anonymous-schema-6>"
+                    },
+                    "name": {
+                      "type": "string",
+                      "x-parser-schema-id": "<anonymous-schema-7>"
+                    }
+                  },
+                  "x-parser-schema-id": "manufactor",
+                  "x-parser-schema-id-level": 1
+                },
+                "x-parser-schema-id": "streetLightHolderArray",
+                "x-parser-schema-id-level": 2
+              },
+              "lightbulb": {
+                "allOf": [
+                  {
+                    "type": "object",
+                    "properties": {
+                      "country": {
+                        "type": "string",
+                        "x-parser-schema-id": "<anonymous-schema-6>"
+                      },
+                      "name": {
+                        "type": "string",
+                        "x-parser-schema-id": "<anonymous-schema-7>"
+                      }
+                    },
+                    "x-parser-schema-id": "manufactor",
+                    "x-parser-schema-id-level": 1
+                  },
+                  {
+                    "type": "object",
+                    "properties": {
+                      "ecologically": {
+                        "type": "string",
+                        "enum": [
+                          "A+++",
+                          "A++",
+                          "A+",
+                          "A",
+                          "B",
+                          "C",
+                          "D",
+                          "E",
+                          "F"
+                        ],
+                        "x-parser-schema-id": "<anonymous-schema-9>"
+                      }
+                    },
+                    "x-parser-schema-id": "<anonymous-schema-8>"
```

Comment:
Code generation would generate an "class" here, which would have an anonymous schema id.

Reply:
Edge case: allOf child enum. Should be something like.
Could be something like: streetLightLightbulbEcologically but i am not sure if a as code owner would be happy with the name. IF some one would like to have that being solved. That there needs be be another PR.